test/app/presenter/WhenSquareWasClicked.ts

```diff
@@ -0,0 +1,31 @@
+import '@testing-library/jest-dom';
+import { ChessBoardView } from '../../../src/app/view/ChessBoardView';
+import { ChessModel } from '../../../src/app/model/ChessModel';
+import { ChessBoardPresenter } from '../../../src/app/presenter/ChessBoardPresenter';
+import { ViewEventBus } from '../../../src/app/view/events/ViewEventBus';
+import { ViewEvent } from '../../../src/app/view/events/ViewEvent';
+import { SquareWasClicked } from '../../../src/app/view/events/SquareWasClicked';
+import { InMemoryViewEventBus } from '../../../src/app/view/events/InMemoryViewEventBus';
+
+describe('SelectedSquare', () => {
+  const viewEvents: ViewEventBus = new InMemoryViewEventBus();
+  const view: ChessBoardView = chessBoardViewMock(viewEvents);
+  const model: ChessModel = {};
+  const presenter: ChessBoardPresenter = new ChessBoardPresenter(view, model);
+
+  it('when square was clicked on the view, then selected piece should be shown', () => {
+    viewEvents.publish(new SquareWasClicked({ x: 1, y: 1 }));
```

Comment:
Test presentera sprawdza czy jesli widok opublikowal wlasciwy event (nie mowi jak to sie dzieje) to czy presenter kazal wyswietlic widokowi to co powinno sie zadziac.